lib/node/rpc.js

```diff
@@ -2152,16 +2152,21 @@ class RPC extends RPCBase {
   }
 
   async estimateFee(args, help) {
-    if (help || args.length !== 1)
-      throw new RPCError(errs.MISC_ERROR, 'estimatefee nblocks');
+    if (help || args.length < 1 || args.length > 2)
+      throw new RPCError(errs.MISC_ERROR, 'estimatefee nblocks (covenantType)');
 
     const valid = new Validator(args);
     const blocks = valid.u32(0, 1);
+    const val = valid.str(1);
 
     if (!this.fees)
       throw new RPCError(errs.MISC_ERROR, 'Fee estimation not available.');
 
-    const fee = this.fees.estimateFee(blocks, false);
+    const type = rules.types[val];
+    if (!type)
```

Comment:
bug: `rules.types: { [string]: number }` and `rules.types.NONE === 0` is falsey.

fix: `rules.types.hasOwnProperty(type)`

Reply:
Good catch. Another way to do this would be `if (typeof type !== number)`, so that it also checks on the type of `type`.

Reply:
Good point. Updated to check `typeof type !== number`